lib/facter/kmod.rb

```diff
@@ -24,7 +24,7 @@
             next unless File.readable?("/sys/module/#{directory}/parameters/#{param}")
 
             begin
-              kmod[directory]['parameters'][param] = File.read("/sys/module/#{directory}/parameters/#{param}").chomp
+              kmod[directory]['parameters'][param] = File.read("/sys/module/#{directory}/parameters/#{param}").chomp.delete("\u0000")
```

Comment:
From my tests, I only see "ASCII text" values, empty files, or files consisting of a single `\n` in parameters.  So any binary data is probably invalid?

In this case, [String#ascii_only?](https://ruby-doc.org/core-3.1.2/String.html#method-i-ascii_only-3F) should help detect this abnormal condition and raise an exception that we can catch below to ignore this parameter completely.

This is probably more consistent than striping arbitrary binary chars, and it shows the intent without the need to git-blame the code to find a commit that reference an issue on GitHub to understand why this was changed (in any case, the "why" should also go in the commit message).

Reply:
Null is valid ASCII though:

```
irb(main):011:0> String.new("\u0000").ascii_only?
=> true
```

Reply:
yes can confirm
```
irb(main):001:0> File.read("/sys/module/virtio_mmio/parameters/device").chomp
=> "\u0000"
irb(main):002:0> File.read("/sys/module/virtio_mmio/parameters/device").chomp.ascii_only?
=> true
```

Reply:
@smortex  As `ascii_only?` won't help here, I'll update the commit message or do you have another idea to resolve this in a better way?

Reply:
Hello, as it seems to me that the solution is now agreed upon, is there a defined timeline to merge this PR?
Is there something I could do to help?
This error is impacting quite a few of our servers, and this PR would be really valuable to us.

Reply:
@prouveyrol if you could test this fix, I think that would help.